frontend/appflowy_flutter/lib/workspace/application/settings/notifications/notification_settings_cubit.dart

```diff
@@ -19,6 +22,16 @@ class NotificationSettingsCubit extends Cubit<NotificationSettingsState> {
           isNotificationsEnabled: _notificationSettings.notificationsEnabled,
         ),
       );
+      getIt<KeyValueStorage>()
+          .getWithFormat(
+            KVKeys.showNotificationIcon,
+            (value) => bool.parse(value),
+          )
+          .then(
+            (value) => state.copyWith(
+              isShowNotificationsIconEnabled: value ?? true,
+            ),
+          );
```

Comment:
Await completion of both and then emit.

You don't even emit the initial value, as you only do `state.copyWith` here.

Reply:
This is how you should do it:
```
Future<void> _initialize() async {
  final settings = await UserSettingsBackendService().getNotificationSettings();
  _notificationSettings = notificationSettings;

  final showNotificationSetting = await getIt<KeyValueStorage>()
        .getWithFormat(KVKeys.showNotificationIcon, (v) => bool.parse(v));

  emit(
    state.copyWith(
      isNotificationsEnabled: _notificationSettings.notificationsEnabled,
      isNotificationsIconEnabled: showNotificationSetting,
    ),
  );
     
  _initCompleter.complete();
}

Reply:
Applied proposal changes, follow [this link](https://github.com/AppFlowy-IO/AppFlowy/pull/5302/commits/b4acc1951f0434ddfac03c4c11383c486fbbd126) to see it